libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m

```diff
@@ -154,40 +154,46 @@ - (void) publishEvents:(NSArray<SFSDKInstrumentationEvent *> *) events {
     }
     @synchronized (self) {
         NSMutableArray<NSString *> *eventIds = [[NSMutableArray alloc] init];
-        BOOL success = YES;
-        NSArray<Class<SFSDKTransform>> *remoteKeySet = [self.remotes allKeys];
-        for (Class<SFSDKTransform> transformClass in remoteKeySet) {
-            if (transformClass) {
-                NSMutableArray<NSDictionary *> *eventsJSONArray = [[NSMutableArray alloc] init];
-                for (SFSDKInstrumentationEvent *event in events) {
-                    [eventIds addObject:event.eventId];
-                    NSDictionary *eventJSON = [transformClass transform:event];
-                    if (eventJSON) {
-                        [eventsJSONArray addObject:eventJSON];
-                    }
-                }
-                Class<SFSDKAnalyticsPublisher> networkPublisher = self.remotes[transformClass];
-                if (networkPublisher) {
-                    BOOL networkSuccess = [networkPublisher publish:eventsJSONArray];
-                    
-                    /*
-                     * Updates the success flag only if all previous requests have been
-                     * successful. This ensures that the operation is marked success only
-                     * if all publishers are successful.
-                     */
-                    if (success) {
-                        success = networkSuccess;
-                    }
+        for (SFSDKInstrumentationEvent *event in events) {
+            [eventIds addObject:event.eventId];
+        }
+        __block BOOL overallSuccess = YES;
+        __block BOOL overallCompletionStatus = NO;
+        __block NSMutableArray<Class<SFSDKTransform>> *remoteKeySet = [[self.remotes allKeys] mutableCopy];
+        __block Class<SFSDKTransform> curTransform = [remoteKeySet objectAtIndex:0];
+        PublishCompleteBlock publishCompleteBlock = ^void(BOOL success) {
+
+            /*
+             * Updates the success flag only if all previous requests have been
+             * successful. This ensures that the operation is marked success only
+             * if all publishers are successful.
+             */
+            if (success) {
+                overallSuccess = success;
```

Comment:
Unless I'm reading this wrong, `overallSuccess` is never _not_ going to be true.